clientconn.go

```diff
@@ -1135,112 +1152,86 @@ func (ac *addrConn) adjustParams(r transport.GoAwayReason) {
 }
 
 func (ac *addrConn) resetTransport() {
```

Comment:
What is the rationale behind this change?

Reply:
Discussed offline.  This behavior is defined by the gRPC docs:

https://github.com/grpc/grpc/blob/master/doc/connectivity-semantics-and-api.md#states-of-connectivity

> channels that receive a GOAWAY when there are no active or pending RPCs should also switch to IDLE to avoid connection overload at servers that are attempting to shed connections.

Reply:
I think as a general philosophy, we try to make functions blocking and have the caller do `go` if needed.

In this particular instance, I _needed_ to do `ac.connect` in a goroutine from `cc.Connect`, as `ac.connect` needs the cc mutex, which needs to be held by `cc.Connect`.

Reply:
Thanks.